packages/core/src/contrib/Constraints.ts

```diff
@@ -181,46 +181,46 @@ const constrDictGeneral = {
   },
 
   /**
-   * Require that shape `s1` overlaps shape `s2` with some padding `padding`.
-   * based on the type of the shape, and with an optional `padding` between them
-   * (e.g. if `s1` should be overlapping `s2` with margin `padding`).
+   * Require that shape `s1` overlaps shape `s2` with some overlap `overlap`.
+   * based on the type of the shape, and with an optional `overlap` between them
+   * (e.g. if `s1` should be overlapping `s2` with margin `overlap`).
    */
   overlapping: (
     [t1, s1]: [string, any],
     [t2, s2]: [string, any],
-    padding = 0.0
+    overlap = 0.0
```

Comment:
@ZiyaoWei Good point; the fact that the `overlap` parameter on the `overlapping` constraint is of type `number` instead of `ad.Num` is actually a mistake, which you can fix via this change:

```suggestion
    overlap: ad.Num = 0.0
```

Then once you do that, it should allow you to make the other changes I mentioned above. (Actually, we really should consistently specify type `ad.Num` in all the places where we provide a numerical default value.)